Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift

```diff
@@ -97,40 +104,62 @@ open class ChatMessageActionsVC: _ViewController, ThemeProvider {
         case nil:
             var actions: [ChatMessageActionItem] = []
 
-            if channelConfig.quotesEnabled {
+            // If a channel is not set, we fallback to using channelConfig only.
+            let canQuoteMessage = channel?.ownCapabilities.contains(.quoteMessage) ?? channelConfig.quotesEnabled
+            let canSendReply = channel?.ownCapabilities.contains(.sendReply) ?? channelConfig.repliesEnabled
+            let canReceiveReadEvents = channel?.ownCapabilities.contains(.readEvents) ?? channelConfig.readEventsEnabled
+            let canUpdateAnyMessage = channel?.ownCapabilities.contains(.updateAnyMessage) ?? false
+            let canUpdateOwnMessage = channel?.ownCapabilities.contains(.updateOwnMessage) ?? true
+            let canDeleteAnyMessage = channel?.ownCapabilities.contains(.deleteAnyMessage) ?? false
+            let canDeleteOwnMessage = channel?.ownCapabilities.contains(.deleteOwnMessage) ?? true
+
+            if canQuoteMessage {
                 actions.append(inlineReplyActionItem())
             }
 
-            if channelConfig.repliesEnabled && !message.isPartOfThread {
+            if canSendReply && !message.isPartOfThread {
                 actions.append(threadReplyActionItem())
             }
 
-            if channelConfig.readEventsEnabled && (!message.isPartOfThread || message.showReplyInChannel) {
+            if canReceiveReadEvents && (!message.isPartOfThread || message.showReplyInChannel) {
                 actions.append(markUnreadActionItem())
             }
 
             actions.append(copyActionItem())
 
-            if message.isSentByCurrentUser {
-                actions += [editActionItem(), deleteActionItem()]
+            if canUpdateAnyMessage {
```

Comment:
nit: feels cleaner with one if with "||", but up-to you.

Reply:
IMO, adding the `||` here can add a little bit more cognitive load, and especially if we add more logic in the future, then it starts to look like a mess. For now I'll keep it this way 👍

Reply:
sure, as I said, up to you 👍